pkg/api/middleware/trustedUrl.go

```diff
@@ -0,0 +1,30 @@
+package middleware
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var MetricTrustedURLRequests = prometheus.NewCounterVec(prometheus.CounterOpts{
+	Name: "http_trusted_url_requests",
+	Help: "Counts the requests with URL queries",
+}, []string{"result"})
+
+// TrustedURL ensures that the `url` query parameter (if it exists) aims at an HTTP or HTTPS website, disallowing e.g. `chrome://` and `file://`.
+func TrustedURL(h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		url := r.URL.Query().Get("url")
```

Comment:
Should we `url.Parse` and check the `Scheme` to avoid any shenanigans?

Reply:
good idea! doing that now :)